plugins/optimization-detective/optimization.php

```diff
@@ -217,6 +214,22 @@ function od_optimize_template_output_buffer( string $buffer ): string {
 	$tag_visitor_context  = new OD_Tag_Visitor_Context( $processor, $group_collection, $link_collection );
 	$current_tag_bookmark = 'optimization_detective_current_tag';
 	$visitors             = iterator_to_array( $tag_visitor_registry );
+
+	/**
+	 * ETag generated for the current environment.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @global string $od_etag
+	 */
+	global $od_etag;
```

Comment:
I think we should avoid adding a global here. I wonder, could the `$current_etag` be a new property on `OD_URL_Metric_Group_Collection`? Then all the groups will have access to it.

Reply:
Good point—makes sense to avoid the global. Updated in cb64e1d.